shell/browser/api/electron_api_session.cc

```diff
@@ -853,6 +853,15 @@ void Session::SetPermissionRequestHandler(v8::Local<v8::Value> val,
          blink::PermissionType permission_type,
          ElectronPermissionManager::StatusCallback callback,
          const base::Value& details) {
+#if (BUILDFLAG(IS_MAC))
+        if (permission_type == blink::PermissionType::GEOLOCATION) {
+          auto* command_line = base::CommandLine::ForCurrentProcess();
```

Comment:
Looks like this duplicates the logic from `IsGeolocationDisabledViaCommandLine`. It's probably worth extracting that function to a place where it can be reused.